src/main/drivers/dshot_bitbang_decode.c

```diff
@@ -0,0 +1,290 @@
+#include <stdint.h>
+#include <string.h>
+#include <stdio.h>
+
+#include "platform.h"
+
+#if defined(USE_DSHOT_TELEMETRY)
+
+#include "common/maths.h"
+#include "common/utils.h"
+#include "drivers/dshot.h"
+#include "drivers/dshot_bitbang_decode.h"
+
+#define MIN_VALID_BBSAMPLES ((21 - 2) * 3)
+#define MAX_VALID_BBSAMPLES ((21 + 2) * 3)
+
+// setting this define in dshot.h allows the cli command dshot_telemetry_info to
+// display the received telemetry data in raw form which helps identify
+// the root cause of packet decoding issues.
+
+#ifdef DEBUG_BBDECODE
+uint16_t bbBuffer[134];
+#endif
+
+
+/* Bit band SRAM definitions */
+#define BITBAND_SRAM_REF   0x20000000
+#define BITBAND_SRAM_BASE  0x22000000
+#define BITBAND_SRAM(a,b) ((BITBAND_SRAM_BASE + ((a-BITBAND_SRAM_REF)<<5) + (b<<2)))  // Convert SRAM address
```

Comment:
Also ,... isn't `a` 16 bit wide (*64 to get bitband address)?

Reply:
![image](https://user-images.githubusercontent.com/18134492/64168608-34316700-ce4c-11e9-8474-2aa0ca548120.png)

Reply:
Yes, you are right ... `a` is already multiplied ...